kubemarine/thirdparties.py

```diff
@@ -79,12 +79,6 @@ def enrich_inventory_apply_defaults(inventory, cluster):
                                     'Expected any of %s, but \'%s\' found.'
                                     % (destination, all_nodes_names, node_name))
 
-        # if source is re-defined by user, but "sha1" is not provided,
-        # then remove default "sha1", because it may be wrong
-        raw_config = raw_inventory.get('services', {}).get('thirdparties', {}).get(destination, {})
-        if 'source' in raw_config and 'sha1' not in raw_config and 'sha1' in config:
-            del config['sha1']
```

Comment:
Is it supported for our default thirdparties (e.g. calicoctl, crictl) that the user can define their own versions? Like, we have `crictl-1.2.3` in our defaults, but user specifies `crictl-1.2.4`. If this is valid, then I think our default SHAs can lead to a situation where user specifies `crictl-1.2.4` (+ we insert default SHA for `crictl-1.2.3`), but at the same time user already has `crictl-1.2.3` on nodes. In this case, we will compare default SHA to the node SHA and they will be equal (1.2.3), so we will successfully say that 1.2.4 is installed.

Reply:
As we discussed, kubemarine should download thirdparty from source and get sha from it.
This way was implemented.